crates/sui-core/src/consensus_handler.rs

```diff
@@ -498,6 +498,19 @@ impl SequencedConsensusTransaction {
             false
         }
     }
+
+    pub fn as_shared_object_txn(&self) -> Option<&SenderSignedData> {
```

Comment:
There is also function `executable_transaction_digest` that does similar match - should we extract this somewhat tricky code into single `as_certificate` and use it both here and in `executable_transaction_digest`?

Reply:
I tried, but it got very awkward because of the mismatch between CertifiedTransaction (for user txns) and VerifiedExecutableTransaction (for system transactions). The only way to share the code was to reduce them to SenderSignedData, but then getting the digest is an uncached operation.